environment/src/main/java/com/sequenceiq/environment/environment/service/recipe/EnvironmentRecipeService.java

```diff
@@ -0,0 +1,70 @@
+package com.sequenceiq.environment.environment.service.recipe;
+
+import static com.sequenceiq.authorization.resource.AuthorizationResourceType.RECIPE;
+
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import javax.inject.Inject;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Service;
+
+import com.sequenceiq.authorization.resource.AuthorizationResourceType;
+import com.sequenceiq.authorization.service.AuthorizationResourceCrnListProvider;
+import com.sequenceiq.cloudbreak.api.endpoint.v4.recipes.RecipeV4Endpoint;
+import com.sequenceiq.cloudbreak.auth.altus.EntitlementService;
+import com.sequenceiq.cloudbreak.recipe.RecipeCrnListProviderService;
+import com.sequenceiq.environment.environment.domain.FreeIpaRecipe;
+import com.sequenceiq.environment.environment.repository.FreeIpaRecipeRepository;
+
+@Service
+public class EnvironmentRecipeService implements AuthorizationResourceCrnListProvider {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(EnvironmentRecipeService.class);
+
+    @Inject
+    private EntitlementService entitlementService;
+
+    @Inject
+    private RecipeV4Endpoint recipeV4Endpoint;
+
+    @Inject
+    private FreeIpaRecipeRepository freeIpaRecipeRepository;
+
+    @Inject
+    private RecipeCrnListProviderService recipeCrnListProviderService;
+
+    public EnvironmentRecipeService(EntitlementService entitlementService, RecipeV4Endpoint recipeV4Endpoint,
+            FreeIpaRecipeRepository freeIpaRecipeRepository, RecipeCrnListProviderService recipeCrnListProviderService) {
+        this.entitlementService = entitlementService;
+        this.recipeV4Endpoint = recipeV4Endpoint;
+        this.freeIpaRecipeRepository = freeIpaRecipeRepository;
+        this.recipeCrnListProviderService = recipeCrnListProviderService;
+    }
+
+    public Set<String> getRecipes(Long environmentId) {
+        return freeIpaRecipeRepository.findByEnvironmentId(environmentId).stream().map(FreeIpaRecipe::getRecipe).collect(Collectors.toSet());
+    }
+
+    public void saveRecipes(Set<String> recipes, Long environmentId) {
+        freeIpaRecipeRepository.saveAll(recipes.stream().map(recipe -> new FreeIpaRecipe(environmentId, recipe)).collect(Collectors.toSet()));
```

Comment:
would it make sense to add log line here? the logger is not used in this class anyway